src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs

```diff
@@ -30,8 +33,37 @@ public StartRpc(INethermindApi api)
         public async Task Execute(CancellationToken cancellationToken)
         {
             IJsonRpcConfig jsonRpcConfig = _api.Config<IJsonRpcConfig>();
+            IKeyStoreConfig keyStoreConfig = _api.Config<IKeyStoreConfig>();
             ILogger logger = _api.LogManager.GetClassLogger();
 
+            if (string.IsNullOrEmpty(jsonRpcConfig.JwtSecretFile))
+            {
+                string oldPath = "keystore/jwt-secret";
+                string newPath = Path.Join(keyStoreConfig.KeyStoreDirectory, "jwt-secret");
+                // check if jwt-secret file already exists in previous default directory
```

Comment:
Logic looks better, now code:
You can assign newPath directly in this line, which will remove the need for an extra if and simplify the code slightly. Additionally, you can move this logic to a private method, such as ConfigureJwtSecret (or a similar appropriately named method

Reply:
But is it okay if the private method takes three parameters as input -> jsonRpcConfig, keyStoreConfig and logger? 

Otherwise, we'll have to create state variables at class level for them.

Reply:
It's okay to pass via function params.